### PR TITLE
Add reboot functionality

### PR DESCRIPTION
**Pull request by robertbaldyga**

### CODE REVIEW

File: connection/ssh_executor.py [hunk 1]
```diff
@@ -76,6 +76,13 @@ def rsync(self, src, dst, delete=False, symlinks=False, exclude_list=[],
     def is_remote(self):
         return True
 
+    def reboot(self):
+        try:
+            self.run("shutdown --reboot now")
+        except Exception:
+            pass
```
Comment (katlapinka) [inline connection/ssh_executor.py]: I think there should be raised some kind of exception, so we can handle it elsewhere and for example make hard reset when shutdown command could not be executed

File: connection/ssh_executor.py [hunk 2]
```diff
@@ -76,6 +76,13 @@ def rsync(self, src, dst, delete=False, symlinks=False, exclude_list=[],
...
+        self.wait_for_connection()
```
Comment (katlapinka) [inline connection/ssh_executor.py]: I think there should be raised some kind of exception, so we can handle it elsewhere and for example make hard reset when shutdown command could not be executed

Comment (Deixx) [reply connection/ssh_executor.py]: If the shutdown for some reason fails, self.wait_for_connection() will be able to connect to DUT and will not wait at all.
I recommend adding a method that waits for loss of connection. Then wait for the connection to get up.

Comment (Deixx) [reply connection/ssh_executor.py]: In below scenario DUT did reboot, but the test performed following steps before the DUT went down.
2019/12/18 15:25:44 INFO:	Reboot
2019/12/18 15:25:44 INFO:	Check if cache is started
2019/12/18 15:25:45 INFO:	Check if core is added
2019/12/18 15:25:45 INFO:	Check if filesystem is mounted
2019/12/18 15:25:45 INFO:	Check if md5 checksum matches
2019/12/18 15:25:46 INFO:	Test cleanup
2019/12/18 15:25:46 INFO:	Unmounting device /dev/cas1-1.
2019/12/18 15:25:46 INFO:	Stop all caches
PASSED

Comment (robertbaldyga) [reply connection/ssh_executor.py]: Fixed.

**Pull request merged**